cypress/integration/mentors.js

```diff
@@ -1,44 +1,68 @@
 const { cy } = global;
 
 describe('Mentor Filtering', () => {
+  before(function(){
+    cy.login();
+    cy.server({urlMatchingOptions: { matchBase: false, dot: true }});
+    cy.route('GET', '/users/current', 'fixture:users/current/get');
+    cy.route('GET', '/mentors?limit=*', 'fixture:mentors/get');
+    cy.visit('/');
+  })
   it('can filter by technology', () => {
-    cy.visit('/')
-      .filterByName('Brent M Clark')
+    cy.filterByName('Brent M Clark')
       .getByTestId('technology-filter-autocomplete')
       .type('reactjs{enter}', { force: true });
-    // .getByTestId('mentor-card')
-    // .should('have.length', 1);
   });
 
   it('can filter by country', () => {
-    cy.visit('/')
-      .filterByName('Brent M Clark')
+    cy.filterByName('Brent M Clark')
       .getByTestId('country-filter-autocomplete')
       .type('United States{enter}', { force: true });
     // .getByTestId('mentor-card')
     // .should('have.length', 1);
   });
 
   it('can filter by name', () => {
-    cy.visit('/').filterByName('Brent M Clark');
+    cy.filterByName('Brent M Clark');
     // .getByTestId('mentor-card')
     // .should('have.length', 1);
   });
 
   it('can filter by language', () => {
-    cy.visit('/')
-      .filterByName('Brent M Clark')
+    cy.filterByName('Brent M Clark')
       .getByTestId('language-filter-autocomplete')
       .type('English{enter}', { force: true });
     // .getByTestId('mentor-card')
     // .should('have.length', 1);
   });
 
   it('can clear filter', () => {
-    cy.visit('/')
-      .filterByName('Brent M Clark')
-      .getByTestId('clear-filter')
-      .click();
+    cy.filterByName('Brent M Clark')
+    cy.get('div.input-container:nth-child(3) div.clear-btn button').click();
```

Comment:
Instead of using a lengthy selector, which makes for a brittle test, how do you feel about going back to using the `clear-filter` button?  If, instead, your goal was to test the per-input clear icon – which is also great! – then this approach would be a bit less brittle:

``` javascript
cy.getByLabelText('Name').within($input => {
  cy.getByText('clear').click()
})
```

The code above is off the top of my head, so it may not be perfect.  It should, however, serve as a hint for how to make that selector less brittle.

Reply:
@brentmclark  Added changes as per your review.